### PR TITLE
Max framework

### PR DESCRIPTION
**Pull request by I-Hong**
This PR is to apply MAX-FRAMEWORK.

### CODE REVIEW

File: api/predict.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+from core.model import ModelWrapper, read_still_image
+from flask_restplus import fields
+from werkzeug.datastructures import FileStorage
+
+from werkzeug.exceptions import BadRequest
+from maxfw.core import MAX_API, PredictAPI
```
Comment (djalova) [inline api/predict.py]: I think we should start organizing our imports
https://www.python.org/dev/peps/pep-0008/#imports

File: core/model.py [hunk 2]
```diff
@@ -9,11 +7,15 @@
 import sys
 from mtcnn.mtcnn import MTCNN
 from PIL import Image
-from keras import backend
+from config import DEFAULT_MODEL_PATH
+from maxfw.model import MAXModelWrapper
```
Comment (djalova) [inline core/model.py]: These imports could be cleaned up too

File: core/model.py [hunk 3]
```diff
@@ -45,6 +54,7 @@ def __init__(self, path=DEFAULT_MODEL_PATH):
         self.stage_num = [3, 3, 3]
         self.lambda_local = 1
         self.lambda_d = 1
+        self.detector = MTCNN()
```
Comment (djalova) [inline core/model.py]: MTCNN is being initialized here the 2nd time.

File: core/model.py [hunk 4]
```diff
@@ -54,24 +64,22 @@ def __init__(self, path=DEFAULT_MODEL_PATH):
...
+        #     print('Wrong input!')
+        #     sys.exit()
 
-        if pyFlag == '3':
-            input_img = cv2.cvtColor(input_img, cv2.COLOR_BGR2RGB)
+        # if pyFlag == '3':
+        #     input_img = cv2.cvtColor(input_img, cv2.COLOR_BGR2RGB)
 
+        ad = 0.4
+        detected = ''  # make this not local variable
```
Comment (djalova) [inline core/model.py]: I'm not sure what you mean with this comment.

File: core/model.py (hunk 3 above)
Comment (I-Hong) [reply core/model.py]: Thanks @djalova, it's been updated.

File: core/model.py (hunk 4 above)
Comment (I-Hong) [reply core/model.py]: Thanks @djalova. removed.

File: api/predict.py (hunk 1 above)
Comment (I-Hong) [reply api/predict.py]: Thanks @djalova , but using `absolute imports` may be more readable.

Comment (djalova) [reply api/predict.py]: I mean the grouping.

File: api/metadata.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from core.model import ModelWrapper
+from maxfw.core import MAX_API, MetadataAPI, METADATA_SCHEMA
+
+
+class ModelMetadataAPI(MetadataAPI):
+
+    @MAX_API.marshal_with(METADATA_SCHEMA)
+
+
```
Comment (djalova) [inline api/metadata.py]: Nit: There's too many spaces between the decorator and the function

File: api/predict.py (hunk 1 above)
Comment (I-Hong) [reply api/predict.py]: Thanks, I've updated it.

File: api/metadata.py (hunk 5 above)
Comment (I-Hong) [reply api/metadata.py]: removed the lines. Thanks @djalova.

File: core/model.py [hunk 6]
```diff
@@ -23,13 +23,20 @@ def draw_label(image, point, label, font=cv2.FONT_HERSHEY_SIMPLEX,
...
-class ModelWrapper(object):
-    """Model wrapper for SavedModel format"""
+class ModelWrapper(MAXModelWrapper):
+    MODEL_META_DATA = {
+        'id': 'ssrnet',
+        'name': 'SSR-Net Facial Age Estimator Model',
+        'description': 'SSR-Net Facial Recognition and Age Prediction model; trained using Keras on the IMDB-WIKI dataset',
+        'type': 'facial-recognition',
+        'license': 'MIT'
+    }
```
Comment (MLnick) [inline core/model.py]: Need to add `source` field here too, see e.g. https://github.com/IBM/MAX-Object-Detector/blob/master/config.py#L34

File: app.py [hunk 7]
```diff
@@ -1,14 +1,7 @@
...
-# load default config
-app.config.from_object('config')
-# load override config if exists
-if 'APP_CONFIG' in os.environ:
-    app.config.from_envvar('APP_CONFIG')
-api.init_app(app)
-
-if __name__ == '__main__':
-    app.run(host='0.0.0.0')
+max_app = MAXApp()
```
Comment (MLnick) [inline app.py]: Need to add `API_TITLE`, `API_DESC`, `API_VERSION` here, see e.g. https://github.com/IBM/MAX-Object-Detector/blob/master/app.py#L5

File: core/model.py (hunk 6 above)
Comment (I-Hong) [reply core/model.py]: Sure, added.

File: app.py (hunk 7 above)
Comment (I-Hong) [reply app.py]: Thanks @MLnick, it's been updated.

**Comment by ajbozarth** [comment]
What's the eta on this? @ptitzler would like to update the long running instance for this but we need a v1.1 release first

**Comment by I-Hong** [comment]
@ajbozarth and @ptitzler , I think it should be okay. Sorry to forget to mention.

File: core/model.py [hunk 8]
```diff
@@ -30,8 +32,15 @@ def read_still_image(image_data):
...
         abort(400, 'Invalid file type/extension. Please provide a valid image (supported formats: JPEG, PNG, TIFF).')
 
-class ModelWrapper(object):
-    """Model wrapper for SavedModel format"""
+class ModelWrapper(MAXModelWrapper):
+    MODEL_META_DATA = {
+        'id': 'ssrnet',
+        'name': 'SSR-Net Facial Age Estimator Model',
+        'description': 'SSR-Net Facial Recognition and Age Prediction model; trained using Keras on the IMDB-WIKI dataset',
+        'type': 'facial-recognition',
```
Comment (MLnick) [inline core/model.py]:
```suggestion
        'type': 'Facial Recognition',
```

File: core/model.py [hunk 9]
```diff
@@ -30,8 +32,15 @@ def read_still_image(image_data):
...
+        'license': 'MIT'
```
Comment (MLnick) [inline core/model.py]: Add in `source` field as per above comment

File: tests/test.py [hunk 10]
```diff
@@ -5,12 +5,12 @@
...
     model_endpoint = 'http://localhost:5000/swagger.json'
     r = requests.get(url=model_endpoint)
+
     assert r.status_code == 200
     assert r.headers['Content-Type'] == 'application/json'
-
     json = r.json()
     assert 'swagger' in json
-    assert json.get('info') and json.get('info').get('title') == 'Model Asset Exchange Server'
+    assert json.get('info') and json.get('info').get('title') == 'Model Asset Exchange Microservice'
```
Comment (MLnick) [inline tests/test.py]:
```suggestion
    assert json.get('info') and json.get('info').get('title') == 'MAX Facial Age Estimator'
```

File: config.py [hunk 11]
```diff
@@ -21,5 +21,6 @@
     'name': 'SSR-Net Facial Age Estimator Model',
     'description': 'SSR-Net Facial Recognition and Age Prediction model; trained using Keras on the IMDB-WIKI dataset',
     'type': 'facial-recognition',
+    'source': 'https://developer.ibm.com/exchanges/models/all/max-facial-age-estimator/',
```
[the remark this answers is not in the document]
Comment (MLnick) [reply config.py]: To be clear - the entire `MODEL_META_DATA` line here can be removed.

[the remark this answers is not in the document]
Comment (I-Hong) [reply config.py]: Thanks @MLnick. Updated.

File: core/model.py (hunk 9 above)
Comment (I-Hong) [reply core/model.py]: Thanks @MLnick .

**Comment by I-Hong** [comment]
Thanks @MLnick, the parts have been updated.

**Pull request merged**